.github/workflows/lighthouse.yml

```diff
@@ -0,0 +1,84 @@
+name: Generate Lighthouse report
+
+on:
+  issue_comment:
+    types: [edited]
+
+jobs:
+  generate_lighthouse_report:
+    timeout-minutes: 30
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout to repo
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 1
+          ref: master
+
+      - name: Add Lighthouse progress comment
+        id: generate_lighthouse_comment
+        uses: marocchino/sticky-pull-request-comment@v2
+        with:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          number: ${{github.event.issue.number}}
+          header: generate_lighthouse_comment
+          message: "⏳ **Generating Lighthouse report...**"
+
+      - name: Capture Vercel preview URL
+        id: vercel_preview_url
+        uses: binary-com/vercel-preview-url-action@v1.0.5
```

Comment:
This action will skip if the Vercel is still deploying, thus there is no need for additional checks here since this step will handle checks for Vercel preview links